pennylane/__init__.py

```diff
@@ -167,6 +179,12 @@ def circuit():
         config (pennylane.Configuration): a PennyLane configuration object
             that contains global and/or device specific configurations.
     """
+    if name not in plugin_devices:
+        # Device does not exist in the loaded device list.
+        # Attempt to refresh the devices, in case the user
+        # installed the plugin during the current Python session.
+        refresh_devices()
```

Comment:
This logic here is the main reason we require a global variable above.

We want to avoid refreshing devices where possible since it can be potentially a slow operation; a global variable allows us to track if the refresh is needed.

Reply:
Is there a way to refresh *only* the device that the user is trying to use? Might provide some speed advantages if it's possible

Reply:
I don't think so no, as far as I can tell `pkg_resources` does all of its magic on import 🙁 

If we wanted this functionality, we could swap to a third party package. The [entrypoints](https://github.com/takluyver/entrypoints) package does exactly this -- it does nothing on import, and instead scans for entrypoints when attempting to load entrypoints. They also have a `entrypoints.get_single()` function which does exactly what you refer to above.